src/dotnet/CarbonAware.DataSources.WattTime/test/WattTimeDataSourceTests.cs

```diff
@@ -112,9 +111,21 @@ public void GetCarbonIntensity_ThrowsWhenRegionNotFound()
         var startDate = new DateTimeOffset(2022, 4, 18, 12, 32, 42, TimeSpan.FromHours(-6));
         var endDate = new DateTimeOffset(2022, 4, 18, 12, 33, 42, TimeSpan.FromHours(-6));
 
-        this.LocationConverter.Setup(l => l.ConvertLocationToBalancingAuthorityAsync(location)).Throws<LocationConversionException>();
+        this.LocationSource.Setup(l => l.ToGeopositionLocationAsync(location)).Throws<LocationConversionException>();
 
         Assert.ThrowsAsync<LocationConversionException>(async () => await this.DataSource.GetCarbonIntensityAsync(new List<Location>() { location }, startDate, endDate));
     }
+    private void SetupBalancingAuthority(BalancingAuthority balancingAuthority, Location location)
+    {
+        this.LocationSource.Setup(r => r.ToGeopositionLocationAsync(location)).Returns(Task.FromResult(location));
+        
+        this.WattTimeClient.Setup(w => w.GetBalancingAuthorityAsync(
+            location.Latitude.ToString() ?? "",
```

Comment:
For the validity of this test, we probably do not want to mock empty string parameters with the same result.  Can we do something like this instead:

```
var latitude = location.Latitude.ToString() ?? throw new Exception();
// ...
this.WattTimeClient.Setup(w => w.GetBalancingAuthorityAsync(latitude, longitude)
// ...
```